SimPEG/potential_fields/magnetics/_numba_functions.py

```diff
@@ -0,0 +1,525 @@
+"""
+Numba functions for magnetic simulation of rectangular prisms
+"""
+import numpy as np
+
+try:
+    import choclo
+except ImportError:
+    # Define dummy jit decorator
+    def jit(*args, **kwargs):
+        return lambda f: f
+
+    choclo = None
+else:
+    from numba import jit, prange
+
+
+def _sensitivity_mag(
+    receivers,
+    nodes,
+    sensitivity_matrix,
+    cell_nodes,
+    regional_field,
+    kernel_x,
+    kernel_y,
+    kernel_z,
+    constant_factor,
+    scalar_model,
+):
+    """
+    Fill the sensitivity matrix for single mag component
+
+    This function should be used with a `numba.jit` decorator, for example:
+
+    ..code::
+
+        from numba import jit
+
+        jit_sensitivity_mag = jit(nopython=True, parallel=True)(_sensitivity_mag)
+
+    Parameters
+    ----------
+    receivers : (n_receivers, 3) array
+        Array with the locations of the receivers
+    nodes : (n_active_nodes, 3) array
+        Array with the location of the mesh nodes.
+    sensitivity_matrix : (n_receivers, n_active_nodes) array
+        Empty 2d array where the sensitivity matrix elements will be filled.
+        This could be a preallocated empty array or a slice of it.
+    cell_nodes : (n_active_cells, 8) array
+        Array of integers, where each row contains the indices of the nodes for
+        each active cell in the mesh.
+    regional_field : (3,) array
```

Comment:
Can you enable it to have a regional field that can be different in each cell? The current forward operator supports this, (though we don't have anything that directly uses it). My hope was to add the option for different sources besides a uniform field (e.g. a static magnetic dipole source, or if it's a very large survey).

This is also the reason the projection direction for `tmi` data was treated separately.

Reply:
I see. That sounds like a nice feature to have. I think it could be a little cumbersome to add it in this PR, specially since we currently don't have a test or example for that runs on the original simulation class. But I would be totally on board of implementing it in a future PR. Would you like to open an issue to capture these thoughts?